CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -1780,6 +1785,14 @@ protected function processSecondaryFinancialTransaction($contactID, &$form, $tem
     );
 
     $result = array();
+
+    if ($tempParams['skipLineItem']) {
+      // We are not processing the line item here because we are processing a membership.
+      // So unset the total_amount param, which stores the non-membership contribution amount.
+      $tempParams['total_amount'] = $membershipContribution->total_amount;
+      $tempParams['contributionID'] = $membershipContribution->id;
+    }
```

Comment:
As far as I understand this will replace `total_amount` and `contributionId` to match the membership one and all remaining params (invoice_id, source etc) are of the non-membership contribution. Not sure but will this not send a mixture of values while making a doPayment() call ?